messagebus/src/main/java/com/yahoo/messagebus/routing/RetryTransientErrorsPolicy.java

```diff
@@ -32,17 +36,22 @@ public RetryTransientErrorsPolicy setEnabled(boolean enabled) {
      * @return This, to allow chaining.
      */
     public RetryTransientErrorsPolicy setBaseDelay(double baseDelay) {
-        this.baseDelay = baseDelay;
+        this.baseDelayUS.set((long)(baseDelay*US));
         return this;
     }
 
     @Override
     public boolean canRetry(int errorCode) {
-        return enabled && errorCode < ErrorCode.FATAL_ERROR;
+        return enabled.get() && errorCode < ErrorCode.FATAL_ERROR;
     }
 
     @Override
     public double getRetryDelay(int retry) {
-        return baseDelay * retry;
+        long retryMultiplier = 0l;
+        if (retry > 1) {
+            retryMultiplier = Math.min(10000, 1L << (retry-1));
```

Comment:
Left shift bit count operand may overflow here with a sufficient number of retries, rolling the multiplier over to the minimum. Consider using `min` on the number of bits shifted by as well.

Reply:
Fixed